packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx

```diff
@@ -70,6 +70,19 @@ export function GridColumnsPanel(props: GridColumnsPanelProps) {
   const ownerState = { classes: rootProps.classes };
   const classes = useUtilityClasses(ownerState);
 
+  const sortedColumns = React.useMemo(() => {
+    if (rootProps.componentsProps?.columnsPanel.sort === 'asc') {
+      return [...columns].sort((a, b) =>
+        (a.headerName || a.field) > (b.headerName || b.field) ? 1 : -1,
+      );
+    } else if (rootProps.componentsProps?.columnsPanel.sort === 'desc') {
```

Comment:
You can remove the `else` because the previous `if` ended with a `return`. Maybe `switch (sort)` will be easier to read

```suggestion
    }
    if (rootProps.componentsProps?.columnsPanel.sort === 'desc') {
```